routes/s3handler.js

```diff
@@ -0,0 +1,239 @@
+import express from 'express'
+import CryptoJS from 'crypto-js'
+import bodyParser from 'body-parser'
+import getObjectURLFromS3 from '../src/utils/s3'
+import faceUtils from '../src/utils/face++'
+
+const clientSecretKey = process.env.AWS_SECRET_ACCESS_KEY
+const expectedMinSize = 0
+const expectedMaxSize = 5000000
+const expectedBucket = 'uploadedphotostomatch'
+const expectedHostname = 'uploadedphotostomatch.s3.amazonaws.com'
+
+const router = express.Router()
+
+// Signs any requests.  Delegate to a more specific signer based on type of request.
+function signRequest(req, res) {
+    if (req.body.headers) {
+        signRestRequest(req, res);
+    }
+    else {
+        signPolicy(req, res);
+    }
+}
+
+// Signs multipart (chunked) requests.
+function signRestRequest(req, res) {
+    var version = req.query.v4 ? 4 : 2,
+        stringToSign = req.body.headers,
+        signature = version === 4 ? signV4RestRequest(stringToSign) : signV2RestRequest(stringToSign);
+
+    var jsonResponse = {
```

Comment:
Might want to avoid `var` in favor of `const` where you can if you don't need to mutate it. If you do, then consider `let` for better block scoping and potentially easier debugging in the future. Might be easier to make these types of changes en masse with ESLint and Google's style guide.

Reply:
👍